### PR TITLE
feat: daemon utoipa

### PR DESCRIPTION
**Pull request by francis2tm**

### CODE REVIEW

File: atoma-daemon/src/handlers/almost_filled_stacks.rs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+pub const ALMOST_FILLED_STACKS_PATH: &str = "/almost_filled_stacks";
+
+#[derive(OpenApi)]
+#[openapi(
+    paths(almost_filled_stacks_fraction_get, almost_filled_stacks_get),
+    components(schemas(Stack))
+)]
+pub(crate) struct AlmostFilledStacksOpenApi;
+
+pub fn almost_filled_stacks_router() -> Router<DaemonState> {
```
Comment (jorgeantonio21) [inline atoma-daemon/src/handlers/almost_filled_stacks.rs:22]: Add docs

File: atoma-daemon/src/handlers/attestation_disputes.rs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        attestation_disputes_against_list,
+        attestation_disputes_against_get,
+        attestation_disputes_own_list,
+        attestation_disputes_own_get
+    ),
+    components(schemas(StackAttestationDispute))
+)]
+pub(crate) struct AttestationDisputesOpenApi;
+
+pub fn attestation_disputes_router() -> Router<DaemonState> {
```
Comment (jorgeantonio21) [inline atoma-daemon/src/handlers/attestation_disputes.rs:27]: Add docs

File: atoma-daemon/src/handlers/nodes.rs [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+        NodeTrySettleStacksResponse,
+        NodeAttestationProofRequest,
+        NodeAttestationProofResponse,
+        NodeClaimFundsRequest,
+        NodeClaimFundsResponse
+    ))
+)]
+pub(crate) struct NodesOpenApi;
+
+pub fn nodes_router() -> Router<DaemonState> {
```
Comment (jorgeantonio21) [inline atoma-daemon/src/handlers/nodes.rs:54]: Add docs

File: atoma-daemon/src/handlers/stacks.rs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+pub const STACKS_PATH: &str = "/stacks";
+
+#[derive(OpenApi)]
+#[openapi(
+    paths(stacks_list, stacks_get, claimed_stacks_list, claimed_stacks_get),
+    components(schemas(Stack, StackSettlementTicket))
+)]
+pub(crate) struct StacksOpenApi;
+
+pub fn stacks_router() -> Router<DaemonState> {
```
Comment (jorgeantonio21) [inline atoma-daemon/src/handlers/stacks.rs:22]: Add docs

File: atoma-daemon/src/handlers/subscriptions.rs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+pub const SUBSCRIPTIONS_PATH: &str = "/subscriptions";
+
+#[derive(OpenApi)]
+#[openapi(
+    paths(subscriptions_list, subscriptions_get),
+    components(schemas(NodeSubscription))
+)]
+pub(crate) struct SubscriptionsOpenApi;
+
+pub fn subscriptions_router() -> Router<DaemonState> {
```
Comment (jorgeantonio21) [inline atoma-daemon/src/handlers/subscriptions.rs:22]: Add docs

File: atoma-daemon/src/handlers/tasks.rs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+use crate::DaemonState;
+
+pub const TASKS_PATH: &str = "/tasks";
+
+#[derive(OpenApi)]
+#[openapi(paths(tasks_list), components(schemas(Task)))]
+pub(crate) struct TasksOpenApi;
+
+pub fn tasks_router() -> Router<DaemonState> {
```
Comment (jorgeantonio21) [inline atoma-daemon/src/handlers/tasks.rs:14]: Add docs

File: atoma-daemon/src/server.rs [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+            shutdown_receiver
+                .changed()
+                .await
+                .expect("Error receiving shutdown signal")
+        });
+    server.await?;
+    Ok(())
+}
+
+/// Creates and configures the main router for the Atoma daemon HTTP API.
```
Comment (jorgeantonio21) [inline atoma-daemon/src/server.rs:117]: Are these docs up to date ?

**Review by jorgeantonio21 — APPROVED** [review]
LGTM

Left a few nitty comments though

**Comment by jorgeantonio21** [comment]
Make sure to have the CI working before merging it

File: atoma-daemon/src/handlers/almost_filled_stacks.rs [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+/// Retrieves all stacks that are filled above a specified fraction threshold for all registered nodes.
+#[utoipa::path(
+    get,
+    path = "/{fraction}",
+    params(
+        ("fraction" = f64, Path, description = "The fraction threshold (0.0 to 100.0) to filter stacks")
+    ),
+    responses(
+        (status = 200, description = "Stack objects that are filled above the specified fraction threshold", body = Vec<Stack>),
```
Comment (Cifko) [inline atoma-daemon/src/handlers/almost_filled_stacks.rs:42]: Can we use `OK` and `INTERNAL_SERVER_ERROR` and similar, instead of the numeric values?

Comment (francis2tm) [reply atoma-daemon/src/handlers/almost_filled_stacks.rs:42]: 100% agreed

Comment (francis2tm) [reply atoma-daemon/src/handlers/almost_filled_stacks.rs:42]: 100% agreed

**Pull request merged**